execution_test.go

```diff
@@ -984,6 +984,59 @@ func TestFederatedQueryFragmentSpreads(t *testing.T) {
 		f.checkSuccess(t)
 	})
 
+	t.Run("with multiple top level fragment spreads (gadget implementation)", func(t *testing.T) {
+		f := &queryExecutionFixture{
+			services: []testService{serviceA, serviceB},
+			query: `
+			query Foo {
+				snapshot(id: "GADGET1") {
+					id
+					name
+					... GadgetFragment
+					... GizmoFragment
+				}
+			}
+
+			fragment GadgetFragment on GadgetImplementation {
+				gadgets {
+					id
+					name
+					agents {
+						name
+						... on Agent {
+							country
+						}
+					}
+				}
+			}
+
+			fragment GizmoFragment on GizmoImplementation {
+				gizmos {
+					id
+					name
```

Comment:
should we have a test for an alias on a field too?

Reply:
The code that deals with aliases is slightly unrelated to fragments as aliases are defined on an `*ast.Field` and here we're testing `*ast.InlineFragment` and `*ast.FragmentSpread`. There are other tests covering correct handling of aliases.

Reply:
Hi @lucianjon 
 in the test case If we fetch either Gadjet Or Gizmo it works. If we want to fetch a list of object where object could be either Gadject or Gizmo it would not work. Can you suggest what can we do in such case

Reply:
@dilip9292 I think if gadget and gizmo have common interface this will not work